core/templates/pages/community-dashboard-page/voiceover-opportunities/voiceover-opportunities.component.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { ContributionOpportunitiesBackendApiService } from
+  // eslint-disable-next-line max-len
+  'pages/community-dashboard-page/services/contribution-opportunities-backend-api.service';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ExplorationOpportunitySummaryObjectFactory } from
+  'domain/opportunity/ExplorationOpportunitySummaryObjectFactory';
+
+describe('Voiceover opportunities component', function() {
+  var ctrl = null;
+  var $rootScope = null;
+  var $scope = null;
+  var contributionOpportunitiesService = null;
+  var explorationOpportunitySummaryObjectFactory = null;
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    $provide.value('ContributionOpportunitiesBackendApiService',
+      TestBed.get(ContributionOpportunitiesBackendApiService));
+    $provide.value('LanguageUtilService', TestBed.get(LanguageUtilService));
+  }));
+
+  beforeEach(function() {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+    });
+
+    explorationOpportunitySummaryObjectFactory = TestBed.get(
+      ExplorationOpportunitySummaryObjectFactory);
+  });
+
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    $rootScope = $injector.get('$rootScope');
+    contributionOpportunitiesService = $injector.get(
+      'ContributionOpportunitiesService');
+
+    spyOn(contributionOpportunitiesService, 'getVoiceoverOpportunities').and
+      .callFake((languageCode, callback) => callback([
+        explorationOpportunitySummaryObjectFactory.createFromBackendDict({
+          id: '1',
+          topic_name: 'topic_1',
+          story_title: 'Story title 1',
+          chapter_title: 'Chapter title 1',
+          content_count: 2,
+          translation_counts: {
+            en: 1
+          }
+        }),
+        explorationOpportunitySummaryObjectFactory.createFromBackendDict({
+          id: '2',
+          topic_name: 'topic_2',
+          story_title: 'Story title 2',
+          chapter_title: 'Chapter title 2',
+          content_count: 4,
+          translation_counts: {
+            en: 2
+          }
+        }),
+      ], true));
+    spyOn(contributionOpportunitiesService, 'getMoreVoiceoverOpportunities').and
+      .callFake((languageCode, callback) => callback([
+        explorationOpportunitySummaryObjectFactory.createFromBackendDict({
+          id: '1',
+          topic_name: 'topic_3',
+          story_title: 'Story title 3',
+          chapter_title: 'Chapter title 3',
+          content_count: 3,
+          translation_counts: {
+            en: 3
+          }
+        }),
+      ], true));
+
+    $scope = $rootScope.$new();
+    ctrl = $componentController('voiceoverOpportunities', {
+      $scope: $scope,
+    });
+    ctrl.$onInit();
+  }));
+
+  it('should initialize correctly controller properties after its' +
```

Comment:
rename this as per suggestion in the naming change PR

Reply:
Done.